### ISSUE TITLE
localstack lambda not honoring create-function timeout argument

### ISSUE THREAD

**Issue by zachschultz** [localstack/localstack#584]
I am creating a function like so:

`awslocal lambda create-function --function-name <MY_FUNC_NAME> --timeout 300 --zip-file fileb://<MY_ZIP>.zip --runtime nodejs6.10  --handler index.handler --role <MY_ROLE>`

and when I run `awslocal lambda list-function` I see the following output

```
{
    "Functions": [
        {
            "Version": "$LATEST",
            "FunctionName": "MY_FUNC_NAME",
            "CodeSize": 50,
            "FunctionArn": "arn:aws:lambda:us-east-1:000000000000:function:MY_FUNC_NAME",
            "Handler": "index.handler",
            "Timeout": 60,
            "Runtime": "nodejs6.10"
        }
  ]
}
```

It seems like even though I am passing in `--timeout 300` it is being ignored

**Comment by whummer** [comment]
Thanks for reporting @zachschultz . This should be fixed in #596, can you please give it a try with the latest snapshot build.

**Comment by dknepprath** [comment]
@whummer, I am able to set the timeout now, but if I don't provide a timeout then no default value is initialized. I will get the following:

```
'Functions': [{'FunctionName': 'myLambdaFunction', 'FunctionArn': 'arn:aws:lambda:us-east-1:000000000000:function:myLambdaFunction', 'Runtime': 'python2.7', 'Handler': 'lambda.handler', 'CodeSize': 50, 'Version': '$LATEST', 'Environment': {}}]
```

I would expect to use the AWS default of 3 seconds. 

I have a coworker who is working on a PR to address the TODO for LambdaContext.get_remaining_time_in_millis. We were thinking of fixing this too.

**Comment by whummer** [comment]
Should be fixed in #1050 - Thanks!
